File: src/WorkflowShell/git/GitLog.rb

class GitLog < Command
  @command_string
  @command_description
  @command_usage

  def initialize
    @command_string = 'gitlog'
    @command_description = 'Displays the git log with each commit on one line in nice colors.'
    @command_usage = @command_string
  end

  def run_command(arguments)
    # Ask the Command class to parse the basic options like help and verbosity
    parser_components = parse_options(arguments)
    if parser_components.basic_options.help
      puts parser_components.basic_options_printer
      exit
    end

    command = "git reflog --pretty=format:'%Cred %H %Cgreen %gD %Cblue %s %C(Yellow) %aN'"
    run_shell_command(command, parser_components.basic_options.verbose)
  end
end

File: src/WorkflowShell/git/GitLog.ts
import Command from "../models/Command";

export default class GitLog extends Command {

  static getString(): string {
    return "glog";
  }

  static getDescription(): string {
    return "Displays the git log with each commit on one line in nice colors.";
  }

  getUsage(): string {
    return GitLog.getString();
  }

  async run(args: any) {
    const ok: boolean = await super.run(args);
    if (!ok) {
      return false;
    }

    const baseCommand: string = "git reflog --pretty=format:'%Cred %H %Cgreen %gD %Cblue %s %C(Yellow) %aN'";
    super.execute(baseCommand);
    return true;
  }
}
